system-test/logging-winston.ts

```diff
@@ -97,55 +88,174 @@ describe('LoggingWinston', () => {
       },
     ];
 
+    const LOG_NAME = 'logging_winston_2_system_tests';
+    const LoggingWinston = inject('../src/index', {
+                             winston: winston2,
+                             'winston/package.json': {version: '2.2.0'}
+                           }).LoggingWinston;
+
+    const logger = new winston2.Logger({
+      transports: [new LoggingWinston({logName: LOG_NAME})],
+    });
+
     it('should properly write log entries', (done) => {
+      const start = Date.now();
       testData.forEach((test) => {
         // logger does not have index signature.
         // tslint:disable-next-line:no-any
         (logger as any)[test.level].apply(logger, test.args);
       });
-      setTimeout(() => {
-        logging.log(LOG_NAME).getEntries(
-            {
-              pageSize: testData.length,
+
+      pollLogs(LOG_NAME, start, testData.length, WRITE_CONSISTENCY_DELAY_MS)
+          .then((entries) => {
+            assert.strictEqual(entries.length, testData.length);
+            entries.reverse().forEach((entry, index) => {
+              const test = testData[index];
+              test.verify(entry);
+            });
+
+            done();
+          });
+    });
+  });
+
+
+  describe('log winston 3', () => {
+    const testTimestamp = new Date();
+    const testData = [
```

Comment:
Would it be possible to share some common form of `testData` for Winston 2 and 3?

Reply:
yeah but i decided instead of concatting the last 2 items on a common test data it was just time better spent elsewhere. ill look again